src/zeek-affinity.h

```diff
@@ -0,0 +1,15 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
```

Comment:
I'll add a comment in the source file to explain why it's independence is warranted:

> // This is all in its own source file primarily because the Linux
// implementation uses the _GNU_SOURCE feature test macro which must be
// defined before including any header file and lumping this together with
// other util functions makes that requirement less apparent and less
// self-contained.

Doesn't necessarily need an independent header file, but seemed the most logical way to organize and also think this function is fine to expose publicly, e.g. for external plugin use if they want.  Let me know if you had other ideas.